packages/pigeon/lib/cpp_generator.dart

```diff
@@ -889,23 +929,28 @@ const flutter::StandardMessageCodec& ${api.name}::GetCodec() {
       }
     });
 
-    indent.newln();
-    indent.format('''
-EncodableValue ${api.name}::WrapError(std::string_view error_message) {
-\treturn EncodableValue(EncodableList{
-\t\tEncodableValue(std::string(error_message)),
-\t\tEncodableValue("Error"),
-\t\tEncodableValue()
-\t});
-}
-EncodableValue ${api.name}::WrapError(const FlutterError& error) {
-\treturn EncodableValue(EncodableList{
-\t\tEncodableValue(error.code()),
-\t\tEncodableValue(error.message()),
-\t\terror.details()
-\t});
-}''');
-    indent.newln();
+    _writeFunctionDefinition(indent, 'WrapError',
+        scope: api.name,
+        returnType: 'EncodableValue',
+        parameters: <String>['std::string_view error_message'], body: () {
+      indent.format('''
+return EncodableValue(EncodableList{
```

Comment:
I didn't fully unwind this into individual structured calls since I was mostly sticking to the functions themselves, but I can add that if you want.